lib/TextDecoder-impl.js

```diff
@@ -0,0 +1,45 @@
+"use strict";
+
+const { labelToName, isSupported, decode } = require("./whatwg-encoding");
+
+exports.implementation = class TextDecoderImpl {
+  constructor(globalObject, constructorArgs = []) {
+    const label = constructorArgs[0];
+    const options = constructorArgs[1] || {};
+
+    const encoding = labelToName(label);
+    if (!isSupported(encoding) || encoding === "replacement") {
+      throw new RangeError(`"${encoding}" is not a supported encoding name`);
+    }
+    this._encoding = encoding;
+    this._errorMode = options._fatal ? "fatal" : "replacement";
+    this._ignoreBOM = options._ignoreBOM || false;
```

Comment:
```suggestion
    this._errorMode = options.fatal ? "fatal" : "replacement";
    this._ignoreBOM = options.ignoreBOM;
```

I don't believe the IDL converter adds the `_` to dictionary members.